common/src/main/java/dev/isxander/yacl3/gui/controllers/dropdown/DropdownMode.java

```diff
@@ -0,0 +1,16 @@
+package dev.isxander.yacl3.gui.controllers.dropdown;
```

Comment:
should be part of the public API, not gui.
you could probably just turn this into two booleans, since it's more api safe in the future, like `strictMode` and `allowEmpty` in the builder.

Reply:
I had two boolean flags at first, but that made the API very confusing. What happens when both of those are set to `true`?

Reply:
> I had two boolean flags at first, but that made the API very confusing. What happens when both of those are set to `true`?

I'd just throw an exception on build if both are true with `Validate` class

Reply:
Actually, I found a clean way to use the booleans that makes obvious sense. "any value" means "any value except empty string", so the two can co-exist. Because "have any string that is not empty" may be something a mod author may reasonably need somewhere.